src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionHelper.cs

```diff
@@ -38,85 +38,97 @@ public static string GetExpressionText(LambdaExpression expression, ExpressionTe
-                    if (name.Contains("__"))
+                        part = binaryExpression.Left;
+                    }
+                    else if (part.NodeType == ExpressionType.MemberAccess)
+                    {
+                        var memberExpressionPart = (MemberExpression)part;
+                        var name = memberExpressionPart.Member.Name;
+
+                        // If identifier contains "__", it is "reserved for use by the implementation" and likely compiler-
+                        // or Razor-generated e.g. the name of a field in a delegate's generated class.
+                        if (name.Contains("__"))
+                        {
+                            // Exit loop. Should have the entire name because previous MemberAccess has same name as the
+                            // leftmost expression node (a variable).
+                            break;
+                        }
+
+                        builder.Insert(0, name);
+                        builder.Insert(0, '.');
+                        part = memberExpressionPart.Expression;
+                    }
+                    else if (part.NodeType == ExpressionType.Parameter)
+                    {
+                        // Exit loop. Have the entire name because the parameter cannot be used as an indexer; always the
+                        // leftmost expression node.
+                        break;
+                    }
+                    else
                     {
-                        // Exit loop. Should have the entire name because previous MemberAccess has same name as the
-                        // leftmost expression node (a variable).
                         break;
                     }
-
-                    nameParts.Push("." + name);
-                    part = memberExpressionPart.Expression;
                 }
-                else if (part.NodeType == ExpressionType.Parameter)
+
+                // If parts start with "model", then strip that part away.
+                if (part == null || part.NodeType != ExpressionType.Parameter)
                 {
-                    // When the expression is parameter based (m => m.Something...), we'll push an empty
-                    // string onto the stack and stop evaluating. The extra empty string makes sure that
-                    // we don't accidentally cut off too much of m => m.Model.
-                    nameParts.Push(string.Empty);
-
-                    // Exit loop. Have the entire name because the parameter cannot be used as an indexer; always the
-                    // leftmost expression node.
-                    break;
+                    var text = builder.ToString();
+                    if (text.StartsWith(".model", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // 6 is the length of the string ".model".
+                        builder.Remove(0, 6);
+                    }
```

Comment:
`builder.Replace(".model", string,Empty, 0, 6);` to replace above block, avoiding the `ToString()`.